Assets/Scripts/Editor/BeatMappingWindow.cs

```diff
@@ -0,0 +1,198 @@
+
+            _songList = new ListView();
+            rootVisualElement.Add(_songList);
+
+            _songList.makeItem = () => new Label();
+            _songList.bindItem = (item, index) => { (item as Label).text = allObjects[index].name; };
+            _songList.itemsSource = allObjects;
+
+            _songList.selectionChanged += OnSongIndexChanged;
+            _songList.style.paddingTop = 30;
+            _songList.style.backgroundColor = new Color(.44f, .44f, .44f, .1f);
+        }
+
+        /// <summary>
+        /// Create object field of type AudioClip.
+        /// </summary>
+        private void CreateSoundClipField()
+        {
+            _soundClipField = new ObjectField("Selected Song");
+            _soundClipField.SetEnabled(false);
+
+            _soundClipField.objectType = typeof(AudioClip);
+            rootVisualElement.hierarchy.Add(_soundClipField);
+        }
+
+        /// <summary>
+        /// Creates basic audio control interface.
+        /// </summary>
+        private void CreateAudioControlInterface()
+        {
+            _volumeSlider = new Slider("Volume Slider", 0, 1);
+            _volumeSlider.value = _waveformDrawer.audioSource.volume;
+
+            _scrubAmoundField = new FloatField("Time Scrub Amount");
+            _scrubAmoundField.value = _defaultTimeScrubAmount;
+            _waveformDrawer.timeScrubAmount = _scrubAmoundField.value;
+
+            _keyBindsButton = new Button(() => { _waveformDrawer.LogKeybinds(); });
+            _keyBindsButton.text = "Keybinds";
+
+            _importTimeStampsButton = new Button(() => { _timeStamper.TryImportTimeStamps(); });
+            _importTimeStampsButton.text = "Import Time Stamps";
+
+            rootVisualElement.Add(_volumeSlider);
+            rootVisualElement.Add(_scrubAmoundField);
+            
+            rootVisualElement.Add(_keyBindsButton);
+            rootVisualElement.Add(_importTimeStampsButton);
+        }
+
+        /// <summary>
+        /// Callback method which is called when user clicks on one of the song names in the list view.
+        /// </summary>
+        /// <param name="obj">Selected audioclip</param>
+        private void OnSongIndexChanged(IEnumerable<object> obj)
+        {
+            List<object> objectList = obj.ToList();
+            if (objectList.Count == 0) //check if selected song is not empty
+                return;
+
+            AudioClip audioClip = (AudioClip)objectList[0];
+            if (audioClip == null) //check if cast to audio clip was valid
+                return;
+
+            _soundClipField.value = audioClip;
+            if (Application.isPlaying)
+            {
+                if (_spectrumDrawer != null) //tell spectrum data to draw the waveform of the selected audio clip
+                    _spectrumDrawer.VisualizeSongSpectrum(audioClip);
+            }
+            else
+            {
+                PlayModeWarning();
+            }
```

Comment:
Remove else statement and return in the if statement above.
```suggestion
            PlayModeWarning();
```